public/app/features/alerting/unified/components/rules/RulesFilter.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { FormEvent, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { Button, Icon, Input, Label, RadioButtonGroup, useStyles } from '@grafana/ui';
+import { GrafanaTheme, SelectableValue } from '@grafana/data';
+import { css, cx } from '@emotion/css';
+import { omitBy, isUndefined } from 'lodash';
+
+import { PromAlertingRuleState } from 'app/types/unified-alerting-dto';
+import { useQueryParams } from 'app/core/hooks/useQueryParams';
+import DataSourceSelect from '../DataSourceSelect';
+import { rulesFiltersSlice } from '../../state/reducers';
+import { useUnifiedAlertingSelector } from '../../hooks/useUnifiedAlertingSelector';
+
+const RulesFilter = () => {
+  const dispatch = useDispatch();
+  const [queryParams, setQueryParams] = useQueryParams();
+  const { rulesFilters } = useUnifiedAlertingSelector((state) => state.filters);
+  const { dataSource, alertState, queryString } = rulesFilters;
+  const { setDataSource, setQueryString, setAlertState, clearFilters, replaceFilters } = rulesFiltersSlice.actions;
+  const styles = useStyles(getStyles);
+  const stateOptions = Object.entries(PromAlertingRuleState).map(([key, value]) => ({ label: key, value }));
+
+  useEffect(() => {
+    if (!Object.keys(rulesFilters).length && Object.keys(queryParams).length) {
+      const queryString = queryParams['queryString'] as string | undefined;
```

Comment:
I found a bug where if you delete text from query, it's saved to query params as empty string `''`, then parsed back as `0` lol: https://github.com/grafana/grafana/pull/32825
Imho better to send `null` to `setQueryParams` to remove the `query` item altogehter if its empty.
Also, always cast it to `String` when reading, as apprently it can be a number -.-

Reply:
Good catch and a fun JS bug 😹